src/pages/SignIn/SignIn.js

```diff
@@ -1,7 +1,65 @@
 import React from 'react';
+import { useLocation } from 'react-router-dom';
+import { KAKAO_AUTH_URL } from './OAuth/OAuth';
+import * as S from './SignIn.style';
 
-const SignIn = () => {
-  return <div>SignIn</div>;
+const User = () => {
+  const location = useLocation();
+
+  const handleHref = () => {
+    localStorage.setItem(
+      'before-address',
+      `${location.pathname}${location.search}`
+    );
+    //현재 위치 endpoint를 recoil atom에 담는다
+    window.location.href = KAKAO_AUTH_URL;
+  };
+  return (
+    <S.UserContainer key={LOGIN_FORM_INFO.id}>
+      <S.Title>{LOGIN_FORM_INFO.text}</S.Title>
+      <S.SignBtnArea>
+        <S.Button variant="contained" onClick={handleHref}>
+          <S.KakaoLogo src={LOGIN_FORM_INFO.logo} />
+          {LOGIN_FORM_INFO.contents}
+        </S.Button>
+      </S.SignBtnArea>
+      <S.CouponEvent>회원가입시 받을 수 있는 혜택 🎉</S.CouponEvent>
+      <S.CouponArea>
+        {COUPON_LIST.map(coupon => (
```

Comment:
```suggestion
        {COUPON_LIST.map(({id, imgurl, text}) => (
```
key가 많지않아서 구조분해할당 하더라도 가독성을 해치지않으면 
parameter에서 바로 해주는것도 방법입니다. :-)